asset_dashboard/models.py

```diff
@@ -129,91 +130,39 @@ class Project(models.Model):
     def __str__(self):
         return self.name or ""
 
-    @receiver([post_save, post_delete], sender="asset_dashboard.LocalAsset")
-    def save_zones_and_districts(sender, instance, **kwargs):
-        phase_geoms = LocalAsset.get_aggregated_assets_by_phase(instance.phase)
-        instance.phase.project.update_project_zones(instance, phase_geoms)
-        instance.phase.project.update_project_districts(instance, phase_geoms)
-
-    def update_project_zones(self, instance, phase_geoms):
+    def update_project_zones(self, phase_geoms):
         zones = Zone.objects.filter(boundary__intersects=phase_geoms)
 
         # Delete existing relationships so we can have a fresh start.
-        instance.phase.project.zones.clear()
+        self.zones.clear()
 
         for zone in zones:
-            instance.phase.project.zones.add(zone)
-            instance.phase.project.save()
+            self.zones.add(zone)
+            self.save()
 
-    def update_project_districts(self, instance, phase_geoms):
+    def update_project_districts(self, phase_geoms):
         district_models = [
             ("commissioner_districts", CommissionerDistrict),
             ("senate_districts", SenateDistrict),
             ("house_districts", HouseDistrict),
         ]
 
         for attribute, model in district_models:
-            districts = model.objects.filter(boundary__intersects=phase_geoms)
+            districts = model.objects.filter(boundary__dwithin=(phase_geoms, D(m=1))).filter(boundary__intersects=phase_geoms)
```

Comment:
this is the dwithin bug fix

Reply:
i find it is suprising that this is using less memory than the boundary_intersects.

can you post the districts.query generated SQL for each of these?